### PR TITLE
Fixes migration unit tests

### PR DESCRIPTION
**Pull request by Kagamine**

### CODE REVIEW

**Comment by Kagamine** [comment]
@caleblloyd Did the UT in your machine can pass 43 cases? or you can only pass 7 cases now?

**Comment by caleblloyd** [comment]
I think I need to switch the Connection Settings to do a lazy initialize

**Comment by Kagamine** [comment]
In this way, the migrations will be mitigated now, but on update will be broken. I will keep investigate the next days. I was too busy in previous term to maintain this project. Thanks for your contributions these days.

**Comment by caleblloyd** [comment]
@Kagamine switched `ConnectionSettings` to lazy initialize in https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/commit/2ea2961e2b21604d8fe920d3f1a0bd365815a76a you can merge the dev branch to get the fix.

Now the dev branch UT is:

```
Total tests: 57. Passed: 43. Failed: 13. Skipped: 1.
```

**Comment by Kagamine** [comment]
@caleblloyd Now the UT fails 3 cases only. I want to checkin this and fix them later/

File: src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs [hunk 1]
```diff
@@ -389,10 +390,15 @@ protected override void ColumnDefinition([CanBeNull] string schema, [NotNull] st
...
-                            throw new InvalidOperationException(
-                                $"Error in {table}.{name}: DATETIME does not support values generated " +
-                                "on Add or Update in MySql <= 5.5, try explicitly setting the column type to TIMESTAMP");
+                        try {
+                            if (_options.ConnectionSettings.ServerVersion.SupportsDateTime6)
+                                throw new InvalidOperationException(
+                                    $"Error in {table}.{name}: DATETIME does not support values generated " +
+                                    "on Add or Update in MySql <= 5.5, try explicitly setting the column type to TIMESTAMP");
+                        }
+                        catch {
```
Comment (caleblloyd) [inline src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs]: We should not catch and ignore this. Instead we should mock ConnectionSettings in the UT so it doesn't fail

Comment (Kagamine) [inline src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs]: reverted

**Comment by Kagamine** [comment]
Removed try catch will throw an exception, could you investigate this? @caleblloyd

**Comment by caleblloyd** [comment]
Yes I'll work up a mock options

**Comment by Kagamine** [comment]
@caleblloyd Does pomelo 2.0 not work?

I've got an exception:

Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: contextOptions

**Comment by caleblloyd** [comment]
@Kagamine that's right it does not work yet.  I'm tracking issues with upstream as I hit them in https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/288

**Pull request merged**